zeus/run/dataloader.py

```diff
@@ -741,13 +972,13 @@ def __next__(self):
             # We're done warming up. Start the actual profiling window.
             elif (
                 self.prof_state == WARMING_UP
-                and time.monotonic() - self.warmup_start_time >= self.warmup_sec
+                and self.sample_num - self.warmup_start_sample >= self.warmup_iter
```

Comment:
```suggestion
                and self.sample_num - self.warmup_start_sample == self.warmup_iter
```
Maybe we should raise an error when `self.sample_num - self.warmup_start_sample > self.warmup_iter`? WDYT? Ditto for line 981.

Reply:
👍  Let's try to add it in both locations.